### PR TITLE
(WIP)(MODULES-2207) update modsync - windows

### PR DESCRIPTION
**Pull request by ferventcoder**
Fix sys-admin version to relax it a bit and let Puppet control that or
source override pick those up so there are no issues when it loads.

### CODE REVIEW

File: moduleroot/Gemfile [hunk 1]
```diff
@@ -41,61 +38,60 @@ group <%= group %> do
...
+gem "hiera", *location_for(hieraversion) if hieraversion
 
-# For Windows dependencies, these could be required
-# based on the version of Puppet you are requiring.
-# Anything greater than v3.5.0 is going to have
-# Windows-specific dependencies dictated by the gem
-# itself.
-# The other scenario is when you are faking out the
-# Puppet Gem to use a local file path / git path.
+<% if @configs['supports_windows'] -%>
```
Comment (ferventcoder) [inline moduleroot/Gemfile]: This was added given the thought that this may roll back up to the other modulesync.

File: moduleroot/Gemfile [hunk 2]
```diff
@@ -41,61 +38,60 @@ group <%= group %> do
...
 <% if gem['gem'] == 'beaker-rspec' -%>
-  if beaker_version = ENV['BEAKER_VERSION']
-    gem 'beaker', *location_for(beaker_version)
-  end
   if beaker_rspec_version = ENV['BEAKER_RSPEC_VERSION']
     gem 'beaker-rspec', *location_for(beaker_rspec_version)
   else
     gem 'beaker-rspec',  :require => false
   end
+<% elsif gem['gem'] == 'beaker' -%>
```
Comment (ferventcoder) [inline moduleroot/Gemfile]: This is more deterministic about beaker vs just having beaker-rspec as the choice that puts in both gems.

File: moduleroot/Gemfile [hunk 3]
```diff
@@ -41,61 +38,60 @@ group <%= group %> do
 <%              'version' => gem['version'], -%>
 <%              'git'     => gem['git'], -%>
 <%              'branch'  => gem['branch'], -%>
-<%              'length'  => gem['gem'].length + (("', '".length if gem['version']) || 0) + gem['version'].to_s.length -%>
+<%              'platforms'  => gem['platforms'], -%>
+<%              'length'  => gem['gem'].length + (("', '".length if gem['version']) || 0) + gem['version'].to_s.length, -%>
+<%              'condition'      => gem['condition'] -%>
```
Comment (ferventcoder) [inline moduleroot/Gemfile]: if conditions for gems

File: moduleroot/Gemfile [hunk 4]
```diff
@@ -41,61 +38,60 @@ group <%= group %> do
 <%              'version' => gem['version'], -%>
 <%              'git'     => gem['git'], -%>
 <%              'branch'  => gem['branch'], -%>
-<%              'length'  => gem['gem'].length + (("', '".length if gem['version']) || 0) + gem['version'].to_s.length -%>
+<%              'platforms'  => gem['platforms'], -%>
```
Comment (ferventcoder) [inline moduleroot/Gemfile]: platforms for gems, like `:ruby`

File: moduleroot/Gemfile (hunk 1 above)
Comment (cyberious) [inline moduleroot/Gemfile]: :+1:

File: moduleroot/Gemfile (hunk 2 above)
Comment (ferventcoder) [inline moduleroot/Gemfile]: This is where there will be some rework on the modules currently under modulesync if/when we roll these changes back up.

File: moduleroot/Gemfile (hunk 4 above)
Comment (cyberious) [inline moduleroot/Gemfile]: Line 44 was removed? It was used to keep things lined up, any reason?

File: managed_modules.yml [hunk 5]
```diff
@@ -1,4 +1,6 @@
 ---
+- puppetlabs-acl
+- puppetlabs-dsc
```
Comment (cyberious) [inline managed_modules.yml]: Should also contain sqlserver?

File: moduleroot/Gemfile (hunk 4 above)
Comment (cyberious) [inline moduleroot/Gemfile]: Never mind see it was moved

File: managed_modules.yml (hunk 5 above)
Comment (ferventcoder) [inline managed_modules.yml]: Added just this morning.

Comment (ferventcoder) [inline managed_modules.yml]: Fixed

**Pull request merged**